types/client.d.ts

```diff
@@ -1,28 +1,16 @@
+import { Options } from "./options";
 import { Person } from "./person";
 import { Project, ProjectCreate, ProjectPatch } from "./project";
 
-export interface Options {
-    readonly fields?: Array<string>,
-    readonly eager?: Boolean,
-    readonly eagerL?: Boolean,
-    readonly map?: Boolean,
-    readonly rules?: Boolean,
-    readonly meta?: Boolean,
-    readonly build?: Boolean,
-    readonly fill?: Boolean,
-    readonly resolveA?: Boolean,
-    readonly skip?: Number,
-    readonly limit?: Number,
-    readonly sort?: Array<Object>,
-    readonly raiseE?: Boolean,
-}
-
 export interface Client {
     readonly id: string;
     readonly name: string;
-    readonly website: string;
-    readonly accountManager: Person;
-    readonly clientRepresentative: Person;
+    readonly website?: string;
+    readonly accountManager?: Person;

```

Comment:
In the `Client` model in Registry, it says `accountManager` is a string, the email of the `Person` but not the `Person` model itself. Are you sure you can replace it by the `Person` entity?

Reply:
Yap, because this accountManager references the person, as in, it is the email of the Person.

Reply:
But who fetches the Person given the email? Say I try to use a the ClientAPI, can I pass in the e-mail? If so, "who" gets the Person with that email? Maybe im confused

Reply:
Basically what we want to say here is that the `accountManager` points to the Person, meaning that the e-mail there . There is still no validation to see if the e-mail given to the accountManager field is a valid Person, but maybe later it will be added.
Still, I believe in the future this validation will be made and, for example, if you get a client, you will receive information of the accountManager Person entity (its email, name, etc), like a mongoose `populate`, or, if not, you can get it with the `getPerson` method.

Reply:
I was questioning more from the POV of the TypeScript transpiler. If he sees a `string` being passed into a a `Person` type field, will he allow it?

Reply:
You're right, it won't. We don't notice this problem because the project that is currently using this API does not use TS. It will be a problem if later one uses.